src/version.rs

```diff
@@ -1,7 +1,9 @@
 use sys::*;
 
+pub trait VersionOption {}
```

Comment:
Could you explain why the extra trait is necessary instead of using a `V: Version` bound for `Environment`?

Reply:
there is `Version` trait and `NoVersion` type which doesn't implement `Version`trait. `Environment` struct can be defined over both and to have that as a bound a supertype, which I named `VersionOption`, is required where `Version: VersionOption` and `impl VersionOption for NoVersion`.

Reply:
As per ODBC standard it must be possible to create `Environment` with no version defined so this is why a super trait is required to envelop all the possible types an `Environment` can be defined over. `VersionOption` is any type that implements `Version` or `NoVersion`